spec/facility_spec.rb

```diff
@@ -1,26 +1,120 @@
-      expect(@facility.phone).to eq('(720) 865-4600')
-      expect(@facility.services).to eq([])
-    end
+  it 'can initialize' do
+    expect(@facility).to be_an_instance_of(Facility)
+    expect(@facility.name).to eq('DMV Tremont Branch')
+    expect(@facility.address).to eq('2855 Tremont Place Suite 118 Denver CO 80205')
+    expect(@facility.phone).to eq('(720) 865-4600')
+    expect(@facility.services).to eq([])
   end
+end
 
-  describe '#add service' do
+  describe '#add service' do  
     it 'can add available services' do
       expect(@facility.services).to eq([])
-      @facility.add_services('New Drivers License')
-      @facility.add_services('Renew Drivers License')
-      @facility.add_services('Vehicle Registration')
+      @facility.add_service('New Drivers License')
+      @facility.add_service('Renew Drivers License')
+      @facility.add_service('Vehicle Registration')
       expect(@facility.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
     end
   end
+
+  describe '#add service' do 
+    it 'can add available services' do 
+      @facility_1.add_service("Vehicle Registration")
+  
+      expect(@facility_1.services).to eq(["Vehicle Registration"])
+    end
+  end
+
+  describe '#registered_vehicles' do 
+    it "has no registered vehicles" do 
+     expect(@facility_1.registered_vehicles).to eq([])
+    end
+  end
+
+  describe '#collected_fees' do 
+    it "has no collected fees" do 
+      expect(@facility_1.collected_fees).to eq(0)
+    end
+  end
+
+  describe '#register_vehicle' do 
+    it "adds vehicle to registered vehicles" do
+      expect(@cruz.registration_date).to eq nil
+
+      @facility_1.add_service("Vehicle Registration")
+      @facility_1.register_vehicle(@cruz)
+      # require 'pry';binding.pry
+  
+      expect(@facility_1.registered_vehicles).to eq([@cruz])
+    end
+    #should these two be one test? 
+    it "sets registration date of vehicle registered" do 
+      @facility_1.add_service("Vehicle Registration")
+      @facility_1.register_vehicle(@cruz)
+      # require 'pry';binding.pry
+      expect(@cruz.registration_date).to eq(Date.today)
+    end
+
+    it "will not register vehicle unless service offered at facility" do 
+      @facility_2.register_vehicle(@bolt)
+      
+      expect(@facility_2.services).to eq([])
+      expect(@facility_2.registered_vehicles).to eq([])
+      expect(@bolt.registration_date).to eq(nil)
+    end
+  end 
+  
+  describe "#set_plate_type" do 
+  # it "set_plate_type unit test" do ???????
+  #    set_plate_type(@cruz)
+  
+  #    expect(@cruz.plate_type).to eq(:regular)
+  # end
+    it "sets plate type upon registration" do 
+      @facility_1.add_service("Vehicle Registration")
+      @facility_1.register_vehicle(@cruz)
+      @facility_1.register_vehicle(@camaro)
+      @facility_1.register_vehicle(@bolt)
+    #do I need to test this helper method invidually that I am about to create? 
```

Comment:
Yup! Every helper method you make, you will also need to test.

Be sure to test for plate type up above in your `#register_vehicle` method as well. Directly testing the plate type method is your unit test, but you want to make sure that `register_vehicle` includes that method, which will be your integration test. Similarly, check that `register_vehicle` also collects fees, since it calls upon your `collect_fees` helper method.